gpkit/geometric_program.py

```diff
@@ -146,9 +146,10 @@ def solve(self, solver=None, verbosity=1, *args, **kwargs):
 
         result = {}
         # confirm lengths before calling zip
-        assert len(self.varlocs) == len(solver_out['primal'])
+        primal = solver_out['primal']
```

Comment:
@pgkirsch changing this line to `primal = np.ravel(solver_out['primal'])` seems to fix the cvxopt issues for me.